crate/findex_client/src/findex_rest_client.rs

```diff
@@ -115,14 +115,16 @@ impl<const WORD_LENGTH: usize> MemoryADT for FindexRestClient<WORD_LENGTH> {
             return Err(ClientError::RequestFailed(err));
         }
 
-        let guard = {
+        let guard: Option<[u8; WORD_LENGTH]> = {
             let words: Vec<_> = OptionalWords::deserialize(&response.bytes().await?)?.into();
-            words.into_iter().next().ok_or_else(|| {
-                ClientError::RequestFailed(
-                    "Unexpected response from server. Expected 1 word, got None".to_owned(),
-                )
-            })
-        }?;
+            let n = words.len();
+            if n != 1 {
+                return Err(ClientError::RequestFailed(format!(
+                    "Unexpected response from server. Expected 1 word, got {n}"
+                )));
+            }
+            words.first().copied().flatten()
```

Comment:
You checked this will not panic the previous line:
```suggestion
            words[0]
```

Reply:
fixed